saleor/graphql/checkout/mutations.py

```diff
@@ -465,27 +463,22 @@ def perform_mutation(cls, _root, info, checkout_id):
         taxes = get_taxes_for_checkout(checkout, info.context.taxes)
         clean_checkout(checkout, taxes, info.context.discounts)
 
+        payment = checkout.get_last_active_payment()
+
         try:
+            # FIXME: we shouldn't create the order, but only do checks
```

Comment:
I think this is fine as it is - the same happens in the old storefront.

Reply:
The issue is that if the payment fails, the order is placed anyway. Is this really the expected behavior?

Reply:
We have an issue for that: https://github.com/mirumee/saleor/issues/3877

Reply:
So we'll handle that anyway but we can do it in a separate PR.